react/features/dial-out/components/DialOutNumbersForm.web.js

```diff
@@ -0,0 +1,319 @@
+        const dialOutCodes = this.props._dialOutCodes;
+
+        if (dialOutCodes) {
+            this._setDefaultCode(dialOutCodes);
+        } else {
+            this.props.updateDialOutCodes();
+        }
+    }
+
+    /**
+     * Monitors for dial out code updates and sets a default code to display in
+     * the dropdown trigger if not already set.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.selectedCountry && nextProps._dialOutCodes) {
+            this._setDefaultCode(nextProps._dialOutCodes);
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t, _dialOutCodes } = this.props;
+
+        const items
+            = _dialOutCodes ? this._formatCountryCodes(_dialOutCodes) : [];
+
+        return (
+            <div className = 'form-control'>
+                { this._createDropdownMenu(items) }
+                <div className = 'dial-out-input'>
+                    <input
+                        className = 'input-control'
+                        label = ''
+                        onChange = { this._onInputChange }
+                        placeholder = { t('dialOut.enterPhone') } />
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Creates a {@code StatelessDropdownMenu} instance.
+     *
+     * @param {Array} items - The content to display within the dropdown.
+     * @returns {ReactElement}
+     */
+    _createDropdownMenu(items) {
+        const { code, dialCode } = this.state.selectedCountry;
+
+        return (
+            <StatelessDropdownMenu
+                isOpen = { this.state.isDropdownOpen }
+                items = { [ { items } ] }
+                onItemActivated = { this._onSelect }
+                onOpenChange = { this._onOpenChange }
+                shouldFitContainer = { true }>
+                { this._createDropdownTrigger(dialCode, code) }
+            </StatelessDropdownMenu>
+        );
+    }
+
+    /**
+     * Creates a React {@code Component} with a readonly HTMLInputElement as a
+     * trigger for displaying the dropdown menu. The {@code Component} will also
+     * display the currently selected number.
+     *
+     * @param {string} dialCode - The +xx dial code.
+     * @param {string} countryCode - The country 2 letter code.
+     * @private
+     * @returns {ReactElement}
+     */
+    _createDropdownTrigger(dialCode, countryCode) {
+        return (
+            <div className = 'dropdown'>
+                <CountryIcon
+                    className = 'dial-out-flag-icon'
+                    countryCode = { `${countryCode}` } />
+                <input
+                    className = 'input-control dial-out-code'
+                    readOnly = { true }
+                    type = 'text'
+                    value = { dialCode || '' } />
```

Comment:
I'm wondering if you shouldn't use an input here and just use a div but maybe it's just easier to get the styling you want with the existing css if it's an input.